bot/telegram/send_image.go

```diff
@@ -0,0 +1,98 @@
+package telegram
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api/v5"
+
+	"jsin/bot/message_handler"
+	"jsin/logger"
+	"jsin/pkg/constants"
+)
+
+func (b *Bot) SendImage(update tgbotapi.Update, object message_handler.ObjectDTO) error {
+	file := tgbotapi.FileBytes{
+		Name:  object.ObjectKey,
+		Bytes: object.Object,
+	}
+	message, err := b.bot.Send(
+		tgbotapi.PhotoConfig{
+			BaseFile: tgbotapi.BaseFile{
+				BaseChat: tgbotapi.BaseChat{
+					ChatID:           update.Message.Chat.ID,
+					ReplyToMessageID: 0,
+				},
+				File: file,
+			},
+		},
+	)
+	if err != nil {
+		logger.Errorf("===== Send image failed: %+v", err.Error())
+		return err
+	}
+	logger.Infof("Image sent, message id: %d", message.MessageID)
+
+	return nil
+}
+
+func (b *Bot) SendImageByObject(chatID int64, object message_handler.ObjectDTO) error {
+	file := tgbotapi.FileBytes{
+		Name:  object.ObjectKey,
+		Bytes: object.Object,
+	}
+
+	message, err := b.bot.Send(
+		tgbotapi.PhotoConfig{
+			BaseFile: tgbotapi.BaseFile{
+				BaseChat: tgbotapi.BaseChat{
+					ChatID: chatID,
+				},
+				File: file,
+			},
+			Caption: constants.Greeting,
+		},
+	)
+
+	if err != nil {
+		logger.Errorf("===== Send image failed: %+v", err.Error())
+		return err
+	}
+
+	logger.Infof("Image sent, message id: %d", message.MessageID)
+	return nil
+}
+
+func (b *Bot) SendImageRandomDaily(chatID int64) error {
+	currentTime := time.Now().Format(constants.DayFormater)
+	generateContent, err := b.botHandler.HandleMessage(
+		context.Background(),
+		fmt.Sprintf("Today is %s", currentTime))
+	if err != nil {
+		logger.Errorf("===== Handle message failed: %+v", err.Error())
+		return err
+	}
+
+	if generateContent != nil && generateContent.Message != "" {
+		msg := tgbotapi.NewMessage(chatID, generateContent.Message)
+		_, err = b.bot.Send(msg)
+
+		if err != nil {
+			logger.Errorf("===== Send message failed: %+v", err.Error())
+			return err
+		}
+	}
+
+	if generateContent != nil && generateContent.Object != nil {
+		err = b.SendImageByObject(chatID, *generateContent.Object)
+
+		if err != nil {
+			logger.Errorf("===== Send image failed: %+v", err.Error())
+			return err
+		}
+	}
+
+	logger.Infof("Daily good morning message and image sent successfully.")
```

Comment:
this is not going to be daily message, isn't it? And let's embed more info inside this log for more information!